src/workerd/io/compatibility-date.c++

```diff
@@ -32,6 +39,26 @@ struct CompatDate {
     return !(*this < other);
   }
 
+#ifdef _WIN32
```

Comment:
It looks like the code is highly similar between this and the non-windows version of parse() below. Can we restrict the #ifdefs to cover only the actual differences, to make them clearer and for better maintainability?